controllers/flowlogspipeline/flp_objects.go

```diff
@@ -437,15 +413,21 @@ func (b *builder) buildPipelineConfig() ([]config.Stage, []config.StageParam) {
 			TLS:     b.getKafkaTLS(),
 		})
 	} else {
-		b.addTransformStages(&pipeline)
+		err := b.addTransformStages(&pipeline)
+		if err != nil {
+			return nil, nil, err
+		}
 	}
-	return pipeline.GetStages(), pipeline.GetStageParams()
+	return pipeline.GetStages(), pipeline.GetStageParams(), nil
 }
 
 // returns a configmap with a digest of its configuration contents, which will be used to
 // detect any configuration change
-func (b *builder) configMap() (*corev1.ConfigMap, string) {
-	stages, parameters := b.buildPipelineConfig()
+func (b *builder) configMap() (*corev1.ConfigMap, string, error) {
+	stages, parameters, err := b.buildPipelineConfig()
+	if err != nil {
+		return nil, "", err
+	}
 
 	config := map[string]interface{}{
 		"log-level": b.desired.LogLevel,
```

Comment:
`config.ConfigFileStruct` can be used instead of `map[string]interface{}`

Reply:
hmm there are some weird things happening here: `ConfigFileStruct` doesn't have `health`. Seems like the output of confgen isn't exactly the input of flp ?

Reply:
also LogLevel exists in `ConfigFileStruct` but it seems unused (see `config.ParseConfig` => it doesn't fill log-level).
I guess some more cleanup should be done on FLP

Reply:
Looking at the code of FLP's main, it seems to me that FLP accepts the health port only from the command line arguments.
https://github.com/netobserv/flowlogs-pipeline/blob/d7656a24bb2417bfd2fec25f75fb1ee077c1b86e/cmd/flowlogs-pipeline/main.go
If I'm correct, I think it's a bug in FLP that it doesn't allow the health port to be set from the config file.

Reply:
I think it works, as it uses viper which accepts a config file as a substitute to command line args